### PR TITLE
Fixes VSTS Bug 685153: Some C# templates do not work when invoked in

### PR DESCRIPTION
**Pull request by mkrueger**
the editor

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/685153

> [Fixes VSTS #685153](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/685153)

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-7.7

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/ExtensibleTextEditor.cs [hunk 1]
```diff
@@ -542,24 +542,28 @@ public bool IsTemplateKnown (ExtensibleTextEditor instance)
...
 		}
-		
+
 		public bool DoInsertTemplate ()
 		{
-			string shortcut = CodeTemplate.GetTemplateShortcutBeforeCaret (EditorExtension.Editor);
-			foreach (CodeTemplate template in CodeTemplateService.GetCodeTemplatesAsync (EditorExtension.Editor).WaitAndGetResult (CancellationToken.None)) {
+			var doc = view.WorkbenchWindow?.Document ?? IdeApp.Workbench.ActiveDocument;
+			if (doc == null)
+				LoggingService.LogError ("DoInsertTemplate(): Can't find valid document");
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/ExtensibleTextEditor.cs]: return false here?

**Comment by slluis** [comment]
@mkrueger tests are failing

**Comment by mkrueger** [comment]
locally the tests run for me.

**Comment by slluis** [comment]
@monojenkins backport release-7.7

**Pull request merged**